examples/daal/cpp_sycl/source/datasource/row_merged_to_sycl_homogen.cpp

```diff
@@ -0,0 +1,147 @@
+
+/**
+ * <a name="DAAL-EXAMPLE-CPP-ROWMERGEDTOSYCLHOMOGEN"></a>
+ * \example datastructures_usm.cpp
+ */
+
+#include "daal_sycl.h"
+#include "service_sycl.h"
+
+using namespace daal;
+using namespace daal::services;
+using namespace daal::data_management;
+
+using daal::data_management::internal::convertToSyclHomogen;
+using daal::data_management::internal::SyclHomogenNumericTable;
+using daal::services::internal::SyclExecutionContext;
+
+uint32_t generateMinStd(uint32_t x) {
+  constexpr uint32_t a = 16807;
+  constexpr uint32_t c = 0;
+  constexpr uint32_t m = 2147483647;
+  return (a * x + c) % m;
+}
+
+/* Compute correlation matrix */
+NumericTablePtr computeCorrelationMatrix(const NumericTablePtr &table) {
+  using namespace daal::algorithms;
+
+  covariance::Batch<> covAlg;
+  covAlg.input.set(covariance::data, table);
+  covAlg.parameter.outputMatrixType = covariance::correlationMatrix;
+  covAlg.compute();
+
+  return covAlg.getResult()->get(covariance::correlation);
+}
+
+/* Fill the buffer with pseudo random numbers generated with MinStd engine */
+cl::sycl::event generateData(cl::sycl::queue &q, float *deviceData,
+                             size_t nRows, size_t nCols) {
+  using namespace cl::sycl;
+  return q.submit([&](handler &cgh) {
+    cgh.parallel_for<class FillTable>(range<1>(nRows), [=](id<1> idx) {
+      constexpr float genMax = 2147483647.0f;
+      uint32_t genState = 7777 + idx[0] * idx[0];
+      genState = generateMinStd(genState);
+      genState = generateMinStd(genState);
+      for (size_t j = 0; j < nCols; j++) {
+        deviceData[idx[0] * nCols + j] = (float)genState / genMax;
+        genState = generateMinStd(genState);
+      }
+    });
+  });
+}
+
+int main(int argc, char *argv[]) {
+  constexpr size_t nCols = 10;
+  constexpr size_t nRows = 10000;
+
+  for (const auto &deviceDescriptor : getListOfDevices()) {
+    const auto &device = deviceDescriptor.second;
+    if (device.is_host()) {
+      /* Shared memory allocations do not work on host by design */
+      continue;
+    }
+
+    const auto &deviceName = deviceDescriptor.first;
+    std::cout << "Running on " << deviceName << std::endl << std::endl;
+
+    /* Crate SYCL* queue with desired device */
+    cl::sycl::queue queue{device};
+
+    /* Set the queue to default execution context */
+    Environment::getInstance()->setDefaultExecutionContext(
+        SyclExecutionContext{queue});
+
+    SharedPtr<RowMergedNumericTable> mergedTable =
+        RowMergedNumericTable::create();
+
+    std::vector<float *> memChunks;
+
+    for (size_t i = 0; i < 3; i++) {
+      /* Allocate shared memory to store input data */
+      float *dataDevice = (float *)cl::sycl::malloc_shared(
+          sizeof(float) * nRows * nCols, queue.get_device(),
+          queue.get_context());
+      memChunks.push_back(dataDevice);
+      if (!dataDevice) {
+        std::cout << "USM allocation failed on " << deviceName << std::endl;
+        continue;
+      }
```

Comment:
I thought use case is to create row-merged table that consists of **host homogen tables** and than convert it to the SYCL one.

Reply:
Modified